MoreLinq.Test/SkipLastWhileTest.cs

```diff
@@ -0,0 +1,69 @@
+﻿#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2024 Andy Romero (armorynode). All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using System.Collections.Generic;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class SkipLastWhileTest
```

Comment:
Given that `SkipLastWhile` can be optimised for sequences, read-only lists and lists, you will want to test that the behaviour is identical irrespective of the source type. Have a look at some of the other tests to see how this is done, where we use [`TestExtensions.ToSourceKind`](https://github.com/morelinq/MoreLINQ/blob/03a160ceaa0942f46cf585894a3c6e53df040076/MoreLinq.Test/TestExtensions.cs#L87) to help with this.

Reply:
Due to `BreakingCollection` and `BreakingReadOnlyCollection` not being list-like, the test cases using them fail since `IterateSequence` uses a foreach loop, since it will get the enumerator. Is it possible to successfully test for these cases?

Reply:
Got it. I appreciate your patience. 🙂